pkg/controller/logstorage/linseed/esgateway.go

```diff
@@ -0,0 +1,132 @@
+		return err
+	} else if esAdminUserSecret == nil {
+		r.status.SetDegraded(operatorv1.ResourceNotFound, "Waiting for elasticsearch admin secret", nil, reqLogger)
+		return nil
+	}
+
+	// This secret should only ever contain one key.
+	if len(esAdminUserSecret.Data) != 1 {
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "Elasticsearch admin user secret contains too many entries", nil, reqLogger)
+		return nil
+	}
+
+	var esAdminUserName string
+	for k := range esAdminUserSecret.Data {
+		esAdminUserName = k
+		break
+	}
+
+	// Collect the certificates we need to provision ESGW. These will have been provisioned already by the ES secrets controller.
+	cm, err := certificatemanager.Create(r.client, install, r.clusterDomain, request.TruthNamespace())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the Tigera CA", err, reqLogger)
+		return err
+	}
+	gatewayKeyPair, err := cm.GetKeyPair(r.client, render.TigeraElasticsearchGatewaySecret, request.TruthNamespace())
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceNotFound, "Error getting TLS certificate", err, log)
+		return err
+	} else if gatewayKeyPair == nil {
+		r.status.SetDegraded(operatorv1.ResourceNotFound, "es-gateway key pair not yet available", err, log)
+		return err
+	}
+
+	kubeControllersGatewaySecret, kubeControllersVerificationSecret, kubeControllersSecureUserSecret, err := lscommon.CreateKubeControllersSecrets(ctx, esAdminUserSecret, esAdminUserName, r.client, request)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Failed to create kube-controllers secrets for Elasticsearch gateway", err, reqLogger)
+		return err
+	}
+
+	cfg := &esgateway.Config{
+		Installation:               install,
+		PullSecrets:                pullSecrets,
+		TrustedBundle:              trustedBundle,
+		KubeControllersUserSecrets: []*corev1.Secret{kubeControllersGatewaySecret, kubeControllersVerificationSecret, kubeControllersSecureUserSecret},
+		ClusterDomain:              r.clusterDomain,
+		EsAdminUserName:            esAdminUserName,
+		ESGatewayKeyPair:           gatewayKeyPair,
+		UsePSP:                     usePSP,
+		Namespace:                  request.InstallNamespace(),
+		TruthNamespace:             request.TruthNamespace(),
+	}
+
+	esGatewayComponent := esgateway.EsGateway(cfg)
+	if err = imageset.ApplyImageSet(ctx, r.client, variant, esGatewayComponent); err != nil {
+		r.status.SetDegraded(operatorv1.ResourceUpdateError, "Error with images from ImageSet", err, reqLogger)
+		return err
+	}
+
+	// Copy the es admin user secret to the namespace so that es-gateway has access to it. For single-tenant, this is essentially a no-op.
+	// For multi-tenant, this means copying the es-gateway admin secret to the tenant's namespace.
+	// TODO: We shouldn't use the admin secret here, this is just for prototyping purposes. Instead, each tenant's esgw should get its own credentials.
```

Comment:
I like this idea, even though it might be a little annoying that we have this divergence between EE and Cloud operators. But we can just suck it up for a few releases.